### ISSUE TITLE
Predefined ServiceAccounts to improve SSO/RBAC UX

### ISSUE THREAD

**Issue by jessesuen** [akuity/kargo#1368]
# Proposed Feature

One of the first things people should do with their Kargo instance is to configure SSO and stop using the built-in admin user. It will be common for orgs to have SSO user(s) that is equivalent to the admin user. Today, this process is:

1. GLOBAL_SERVICE_ACCOUNT_NAMESPACES: kargo
2. create a `kargo-admin` ServiceAccount
3. create `kargo-admin` ClusterRoleBinding to `kargo-api` ClusterRole
4. annotate kargo-admin ServiceAccount with a SSO annotations

I feel that Kargo Helm chart should have a convenience option to do 1-3 out of the box to reduce the upfront steps and have a standard ServiceAccount that we could refer to in docs.

In addition, we could introduce a `kargo-viewer` built-in ServiceAccount and ClusterRole/ClusterRoleBinding, that is ready to be annotated for giving readonly access to SSO users.

# Motivation

Getting started will be easier.

# Suggested Implementation

Simply vend these resources in our Helm chart, with a helm values option for additional annotations.

**Comment by jessesuen** [comment]
This is what the convenience option would do:

```yaml
apiVersion: v1
kind: ConfigMap
metadata:
  name: kargo-api
  namespace: kargo
data:
  GLOBAL_SERVICE_ACCOUNT_NAMESPACES: kargo

---
apiVersion: v1
kind: ServiceAccount
metadata:
  annotations:
    rbac.kargo.akuity.io/groups: mydevopsgroup
  name: kargo-admin
  namespace: kargo

---
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRoleBinding
metadata:
  name: kargo-admin
roleRef:
  apiGroup: rbac.authorization.k8s.io
  kind: ClusterRole
  name: kargo-api
subjects:
- kind: ServiceAccount
  name: kargo-admin
  namespace: kargo
```

**Comment by jessesuen** [comment]
One question is if we want to reuse the existing `kargo-api` ClusterRole, or introduce a separate `kargo-admin` ClusterRole. I'm thinking we want them separate, even if the roles are identical right now, since they might not be identical in the future.

**Comment by krancour** [comment]
Different seems the way to go to me as well.

**Comment by sennerholm** [comment]
If you can decide what kind of structure you want in the helm values for this I can try to come with a PR, so I can scrap my own implementation :-)

**Comment by krancour** [comment]
@sennerholm my apologies. I am just now noticing your previous comment after already having opened #1548

If you wanted to review validate the approach against what you were already doing, it may be helpful.

**Comment by sennerholm** [comment]
> @sennerholm my apologies. I am just now noticing your previous comment after already having opened #1548
> 
> If you wanted to review validate the approach against what you were already doing, it may be helpful.

Np, I will in the beginning of next week to backport your change to our 0.4 copy and check it out. I think that we may want to have the "kargo-promoter" in the future, as an Viewer with permission to "promote". Any reason why you remove it?

**Comment by krancour** [comment]
> I think that we may want to have the "kargo-promoter" in the future, as an Viewer with permission to "promote". Any reason why you remove it?

It was looking like it didn't fit in with how things are evolving...

API users who authenticate via SSO don't directly have any permissions on the underlying cluster. They get [mapped to an SA via annotations](https://kargo.akuity.io/how-to-guides/managing-user-permissions). This issue is fundamentally about having a convenient way to specify two specific mappings at install time. Map:

* Some users to an admin role, across all projects, so that it's practical to install Kargo with the admin account disabled.

* Most (all?) users to a viewer role that gives them read-only access across all projects.

The `kargo-promoter` Role that existed before was created in the early days of the project and hadn't been touched in some time and as I said, I couldn't see how it fit in:

* The likelihood of deciding at install time to grant some users permission to promote to any Stage in any project seems small.

* The likelihood of granting users permission to promote to Stages in a _specific project_ (or _specific Stages and not others within a specific project_) seems higher.

* If the former `kargo-promoter` role _wasn't_ going to be bound to an SA that's mapped to certain users at install time it started to look useless from the POV of both the installer and API users.

* For those _not_ interacting with Kargo via API -- i.e. using `kubectl`, gitops'ing the Kargo install, or even gitops'ing individual projects, it seemed that establishing more narrowly scoped permissions for either API users or for fellow `kubectl` user wouldn't be a problem.

Bottom line... it no longer seemed like something that really needed to be there right out of the box.
